### PR TITLE
docs: improve installing_composer.rst

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add note about deployment and `--no-dev`

See https://forum.codeigniter.com/showthread.php?tid=84728&pid=405756#pid405756

After:
Screenshot 2023-01-23 9 05 17

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/installation/installing_composer.rst [hunk 1]
```diff
@@ -45,14 +48,14 @@ If you omit the "project-root" argument, the command will create an
...
+.. important:: When you deploy to your production server, don't forget to run the
+    following command::
 
-A sample such installation command, using the default project-root "appstarter"::
+    > composer install --no-dev
 
-    > composer create-project codeigniter4/appstarter --no-dev
+    The above command will removes the Composer packages only for development
+    that are not needed in the production environment. This will greatly reduce
+    the folder size.
```
Comment (samsonasik) [inline user_guide_src/source/installation/installing_composer.rst:58]: folder -> vendor

File: user_guide_src/source/installation/installing_composer.rst [hunk 2]
```diff
@@ -131,11 +134,20 @@ In your project root::
...
     > composer require codeigniter4/framework
 
+.. important:: When you deploy to your production server, don't forget to run the
+    following command::
+
+    > composer install --no-dev
+
+    The above command will removes the Composer packages only for development
+    that are not needed in the production environment. This will greatly reduce
+    the folder size.
```
Comment (samsonasik) [inline user_guide_src/source/installation/installing_composer.rst:144]: folder -> vendor

File: user_guide_src/source/installation/installing_composer.rst (hunk 1 above)
Comment (kenjis) [reply user_guide_src/source/installation/installing_composer.rst]: Thanks! Fixed.

File: user_guide_src/source/installation/installing_composer.rst (hunk 2 above)
Comment (kenjis) [reply user_guide_src/source/installation/installing_composer.rst]: Done.

**Review by samsonasik — APPROVED** [review]

**Pull request merged**